numpy/core/src/private/mem_overlap.c

```diff
@@ -845,8 +847,7 @@ solve_may_have_internal_overlap(PyArrayObject *a, Py_ssize_t max_work)
 {
     diophantine_term_t terms[NPY_MAXDIMS+1];
     npy_int64 x[NPY_MAXDIMS+1];
-    unsigned int nterms;
-    int i, j;
+    unsigned int i, j, nterms;
```

Comment:
Which would make these `int` again

Reply:
That would probably work, but `nterms` gets passed down through several routines and I'd need to audit all of them to make sure nothing could go wrong. This is the safest, quick fix. I do think the overlap code could use a more exhaustive cleanup now that it has had time to settle, but I don't want to do it in this PR.

Reply:
My guess is that no unsigned integers are needed in the code, except possibly `npy_uintp`, and I'm not sure about that as `ptrdiff_t` should serve if present. I'll open an issue for a `mem_overlap` review.

Reply:
See #10296.